src/store/reducers/keymap.ts

```diff
@@ -0,0 +1,135 @@
+import '@ngrx/core/add/operator/select';
+import { Action } from '@ngrx/store';
+
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/switchMap';
+import { Observable } from 'rxjs/Observable';
+
+import { Keymap } from '../../config-serializer/config-items/Keymap';
+import { KeymapActions } from '../actions';
+import { AppState } from '../index';
+
+const initialState: Keymap[] = [];
+
+export default function(state = initialState, action: Action): Keymap[] {
+    switch (action.type) {
+        case KeymapActions.ADD:
+        case KeymapActions.DUPLICATE:
+            let newKeymap: Keymap = new Keymap;
```

Comment:
`let newKeymap: Keymap = new Keymap(action.payload);` should work after you rebase.